framework/src/task_manager/task_manager_register.c

```diff
@@ -51,19 +51,24 @@ int task_manager_register(char *name, int permission, int timeout)
 
 	if (timeout != TM_NO_RESPONSE) {
 		asprintf(&request_msg.q_name, "%s%d", TM_PRIVATE_MQ, request_msg.caller_pid);
+		if (request_msg.q_name == NULL) {
+			TM_FREE(request_msg.data);
+			return TM_OUT_OF_MEMORY;
+		}
 	}
 
 	status = taskmgr_send_request(&request_msg);
 	if (status < 0) {
+		TM_FREE(request_msg.data);
+		if (request_msg.q_name != NULL) {
```

Comment:
in here, request_msg.q_name cannot be null

Reply:
When timeout is not TM_NO_RESPONSE and sending request is failed, it can be NULL.